### PR TITLE
Array Scopes, Error Event Definitions & Scoped Error Bindings

### LINKED ISSUE #20: Adjust the elementTemplates jsonSchema to ensure compatibility (array scopes, error scopes, error event definition binding)
__What should we do?__

As part of the support of the `bpmn:Error` and `camunda:ErrorEventDefinition` in element templates, the schema must be updated to ensure compatibility.

__Why should we do it?__

The next version `v0.3.0` should ensure compatibility with the new errors support.

Child of camunda/camunda-modeler#2070

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #20

* handle array scopes
* allow error event definitions
  * validate for missing errorRef
  * validate for property type ("String", "Hidden", "Dropdown")
* allow error scopes
  * validate for missing id

### CODE REVIEW

File: resources/schema.json [hunk 1]
```diff
@@ -467,6 +467,52 @@
...
+              "$ref": "#/definitions/properties"
+            }
+          }
+        }
+      }
+    },
+    "scopes": {
+      "$id": "#/definitions/scopes",
+      "type": "array",
+      "description": "Special scoped bindings that allow you to configure nested elements",
```
Comment (MaxTru) [inline resources/schema.json:520]: Any reasons why we dont have a `title` property for `scopes` (we have it for the other cases)

File: resources/schema.json [hunk 2]
```diff
@@ -467,6 +467,52 @@
...
+      "items": {
```
Comment (MaxTru) [inline resources/schema.json:521]: Also for `items` => why no `title` + why no `description`?

File: resources/schema.json [hunk 3]
```diff
@@ -467,6 +467,52 @@
...
+        "$id": "#/scopes/item",
+        "type": "object",
+        "properties": {
+          "type": {
+            "$id": "#scopes/item/type",
+            "type": "string",
+            "pattern": "^(camunda|bpmn):"
```
Comment (MaxTru) [inline resources/schema.json:528]: We only allow scopes of type `camunda:Connector` or `bpmn:Error` right? Why not check explicitly for these two?

File: test/spec/validationSpec.js [hunk 4]
```diff
@@ -23,6 +23,10 @@ describe('validation', function() {
 
   function testTemplate(name, file, only = false) {
 
+    if (!file) {
```
Comment (MaxTru) [inline test/spec/validationSpec.js:28]: Nice!

**Review by MaxTru — COMMENTED** [review]
Really nice how well we can extend the schema - JSONSchema is powerful :-)

Only minor comments, apart these.

:rocket:

**Comment by pinussilvestrus** [comment]
Thanks for the review, very good points 👍  I added a comment on top (https://github.com/camunda/element-templates-json-schema/pull/21/commits/c5d310fa75e998980ce71e23fb5c216b4a6d21de) which changes the pattern with a more restrictive rule. Also added the missing `title` & `description` fields.

**Review by MaxTru — APPROVED** [review]
:clap:

**Pull request merged**